Don't skip/drop deprecated events

captain5050 opened the pull request:
Linux perf will not show deprecated events in 'perf list' unless
--deprecated is passed. As such having such events in the json is mostly
harmless. Some metrics rely on deprecated events and so we prefer to
have them. For example, io_bandwidth_read and io_bandwidth_write on CLX
and SKX use UNC_IIO_PAYLOAD_BYTES_IN.MEM_READ.PART1 and
UNC_IIO_PAYLOAD_BYTES_IN.MEM_WRITE.PART1 respectively.

captain5050 committed:
Don't skip/drop deprecated events

Linux perf will not show deprecated events in 'perf list' unless
--deprecated is passed. As such having such events in the json is mostly
harmless. Some metrics rely on deprecated events and so we prefer to
have them. For example, io_bandwidth_read and io_bandwidth_write on CLX
and SKX use UNC_IIO_PAYLOAD_BYTES_IN.MEM_READ.PART1 and
UNC_IIO_PAYLOAD_BYTES_IN.MEM_WRITE.PART1 respectively.

ZhengjunXing committed:
update perf-uncore-events-clx.csv based on CLX uncore JSON file v1.16

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

ZhengjunXing committed:
update perf-uncore-events-skx.csv to SKX uncore JSON file v1.28

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

ZhengjunXing committed:
Fix origin name event duplicate issue

for the following .csv file, the origin name event will be duplicated for 5 times.
now only keep the first one origin name event.

UNC_CHA_TOR_INSERTS.IA_MISS,,,,,,
UNC_CHA_TOR_INSERTS.IA_MISS,LLC_MISSES.UNCACHEABLE,LLC misses - Uncacheable reads (from cpu) ,:filter1=0x40e33,,,
UNC_CHA_TOR_INSERTS.IA_MISS,LLC_MISSES.MMIO_READ,MMIO reads,:filter1=0x40040e33,,,There was a typo that I fixed
UNC_CHA_TOR_INSERTS.IA_MISS,LLC_MISSES.MMIO_WRITE,MMIO writes,:filter1=0x40041e33,,,
UNC_CHA_TOR_INSERTS.IA_MISS,LLC_REFERENCES.STREAMING_FULL,Streaming stores (full cache line),:filter1=0x41833,64,,
UNC_CHA_TOR_INSERTS.IA_MISS,LLC_REFERENCES.STREAMING_PARTIAL,Streaming stores (partial cache line),:filter1=0x41a33,64,,

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

captain5050 committed:
Fix some python style issues

Typing and style related issues.

captain5050 committed:
Add uncore CSV directory flag

It isn't guaranteed that the script will be run in the same directory as
the CSV files so allow a directory to be specified.

captain5050 committed:
Remove broken_extra_metrics that are fixed

Fixed by a95abd7 in https://github.com/intel/perfmon-metrics.

ZhengjunXing0 commented on uncore_csv_json.py:
```diff
@@ -214,10 +214,7 @@ def is_deprecated(j):
-            return "Deprecated" in j and j["Deprecated"] == "1"
-
-        jl += [update(events[x]) for x in sorted(events.keys()) if not skip(events[x]) and not x in added]
+        jl += [update(events[x]) for x in sorted(events.keys()) if not x in added]
```
It's better to add an option for "Deprecated" events instead to add them in default.

captain5050 replied:
For core events there is no option, so this brings about greater consistency. Such an option is unnecessary as perf has a deprecated flag and hides deprecated events by default.

ZhengjunXing0 replied:
Add an option maybe is useful, when the perf "deprecated flag" function has a problem, we can easy to generate the event list without "deprecated" events and quickly backtrace the issue.

captain5050 replied:
It could. I don't see why it'd be here and specific to uncore events. We could write a new "strip deprecated" command for this, it would just json into json and drop any dictionaries containing 'Deprecated' : '1'.

captain5050 replied:
The latest commit adds a command:
remove-deprecated-json.py
that will remove "Deprecated":"1" events and write them to stdout or a specified filename.

ZhengjunXing committed:
update perf-uncore-events-icx.csv to ICX uncore JSON file v1.15

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

ZhengjunXing committed:
update CSV files

update CSV files to follow the rules of commit 451533ec5118
(Fix origin name event duplicate issue)

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

captain5050 committed:
Extend the unsupported filter expressions

Filter expressions need to be in a format that works for perf or get
fixed via the csv file. When generating uncore events for architectures
like ivytown just CHAFilter was insufficient to generate an event list
that worked with perf. Add a notion of two kinds of filter, one for
events that should be dropped the other for the filter should just be
removed.

This is based on discussion with Kan Liang here:
https://lore.kernel.org/lkml/787d208d-accd-99f6-065b-ba1f914dcd25@linux.intel.com/

captain5050 committed:
Add broken extra metrics for haswellx.

Added now the metrics are available on github.

captain5050 committed:
Add tool to create a json with deprecated events removed

The new tool takes an input json filename and an optional output
filename (default is stdout) and writes the input to the output removing
any events that have a "Deprecated":"1".

ZhengjunXing0 commented:
applied, thanks.
